### ISSUE TITLE
rustc-guide: document testing for broader set of targets 

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rust#54393]
We need to document the recommended way to go about testing for broader set of targets (e.g. `asmjs`) other than just throwing it at bors and hoping it works.

Spawned off of [T-compiler meeting](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/subject/weekly.20meeting.202018-09-20/near/134311522) discussion of #54223

Some choice quotes:

> "I was recommended changing the `.travis.yml` to enable other platforms temporarily for the PR testing instead of queuing for bors each time for the other platforms." -- [davidtwco](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/subject/weekly.20meeting.202018-09-20/near/134311671)

 * > "you can manually fiddle with .travis.yml if you want to enable more, just don't enable all at once (go 1-2 at a time)" -- [simulacrum](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/subject/test.20suite/near/131235830)

Examples of PR's that have been doing the "throw it at bors and hope" dance: PR #54004, PR #54223

**Comment by pnkfelix** [comment]
tagging T-compiler because we're the ones who need to learn to do this. tagging T-infra because I'll want them to double-check that whatever steps we document here are actually good practice to follow.

**Comment by pietroalbini** [comment]
cc @kennytm

**Comment by kennytm** [comment]
If it's about rustc-guide the issue should be opened at https://github.com/rust-lang-nursery/rustc-guide/issues?

**Comment by ehuss** [comment]
I started to document how to test strange platforms here: https://rust-lang-nursery.github.io/rustc-guide/tests/intro.html#testing-with-docker-images  However I never finished because I don't know the answers.  I have some additional notes that I've collected since I wrote that (such as better ways to test WASM and ARM), but I wasn't confident enough to extend the documentation.  I'm willing to help if anyone can share how to test certain things.  

I agree this issue should be on the rustc-guide repo.  I thought there was an open issue, but I can't find one.
